modules/data.land/R/clip_and_move_raster_file.R

```diff
@@ -0,0 +1,47 @@
+#' Clip and Move a Raster File
+#'
+#' Clips a raster to a polygon bounding box, optionally masks to polygon, and saves the 
+#' output in the same format as the input.
+#'
+#' @param input_path Character. Path to the input raster file.
+#' @param polygon An `sf` or `SpatVector` object to be used for clipping and masking.
+#' @param out_path Character. Path to save the processed raster.
+#' @return Character. The path to the saved output raster.
+#' @export
+#' @author David LeBauer
+clip_and_move_raster_files <- function(input_path, polygon, out_path, mask = TRUE, overwrite = TRUE) {
+  rast_in <- terra::rast(input_path)
+  # check that input file exists
+  if (!file.exists(input_path)) {
+    PEcAn.logger::logger.error("Input raster file does not exist: ", input_path)
+  }
+  # check that polygon is valid
+  if (!inherits(polygon, c("sf", "SpatVector"))) {
+    PEcAn.logger::logger.error("Polygon must be an sf object or SpatVector")
+  }
+  if (inherits(polygon, "sf")) {
+    # Convert sf object to SpatVector
+    polygon <- terra::vect(polygon)
+  }
+  # Reproject polygon to raster CRS, convert to SpatVector
+  polygon_proj <- sf::st_transform(polygon, crs = terra::crs(rast_in))
+  polygon_vect <- terra::vect(polygon_proj)
```

Comment:
Any reason not to stick with terra operations here?
```suggestion
  polygon_vect <- terra::project(polygon, rast_in)
```

Reply:
good point!